sentry/src/main/java/io/sentry/ConnectionFactory.java

```diff
@@ -0,0 +1,26 @@
+package io.sentry;
+
+import io.sentry.transport.Connection;
+import org.jetbrains.annotations.NotNull;
+
+public final class ConnectionFactory {
+  private ConnectionFactory() {}
+
+  public static @NotNull Connection create(final @NotNull SentryOptions options) {
+    if (classExists("org.asynchttpclient.AsyncHttpClient")) {
+      return NIOConnectionFactory.create(options);
+    } else {
+      return AsyncConnectionFactory.create(options, options.getEnvelopeDiskCache());
+    }
+  }
+
+  private static boolean classExists(String clazz) {
+    try {
+      Class.forName(clazz, false, ClassLoader.getSystemClassLoader());
+      return true;
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+      return false;
+    }
+  }
```

Comment:
reflection on Android is very bad and we already do some + IO, adding one more does not sound a good idea, this slows down the Sdk init which might drop frames on Mobile, wondering if it makes sense to split the whole "transport" dependent package to a different module